### PR TITLE
added example tests for addresses and bls signing

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
Added address examples:
-bech32 to hex
-hex to bech32
-shard ID computation

Added bls signing example

### CODE REVIEW

File: examples/address_test.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHexAddressToBech32Address(t *testing.T) {
```
Comment (iulianpascalau) [inline examples/address_test.go:12]: :+1:

File: examples/address_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	bech32Address := "erd14uqxan5rgucsf6537ll4vpwyc96z7us5586xhc5euv8w96rsw95sfl6a49"
+
+	bech32AddressBytes, err := addressEncoder.Decode(bech32Address)
+	require.NoError(t, err)
+
+	hexEncodedAddress := hex.EncodeToString(bech32AddressBytes)
+	require.Equal(t, "af006ece83473104ea91f7ff5605c4c1742f7214a1f46be299e30ee2e8707169", hexEncodedAddress)
+}
+
+func TestShardOfAddress(t *testing.T) {
```
Comment (iulianpascalau) [inline examples/address_test.go:36]: :+1:

**Review by iulianpascalau — COMMENTED** [review]
will follow-up with another useful test

**Review by iulianpascalau — COMMENTED** [review]
Maybe add this test in `blsSigning_test.go`?
```GO
package examples

import (
	"encoding/hex"
	"encoding/pem"
	"fmt"
	"strings"
	"testing"

	"github.com/ElrondNetwork/elrond-go/crypto"
	"github.com/ElrondNetwork/elrond-go/crypto/signing"
	"github.com/ElrondNetwork/elrond-go/crypto/signing/mcl"
	mclsig "github.com/ElrondNetwork/elrond-go/crypto/signing/mcl/singlesig"
	"github.com/stretchr/testify/require"
)

var blsPemContents = `
-----BEGIN PRIVATE KEY for cb66c844dc64e0cab854997df9b3fd1b1c071ee25e6634e6b95bdb15f0acb38e7c5ac20d02f231a03fde3abb8220951328a7f550915ff9da4a1960c8005d6dfabc50f776bd17433f29e8d0566871380b439024a70dfade593173c6462ad49318-----
NGYwYmRkNWEyMjM3Y2I2MWU0OTU3NjNkOTNjM2E4NTU3NzQyMGZhZjUxZTRiYmI0
MGQ1YTAzZTkxNWYxZTIxZg==
-----END PRIVATE KEY for cb66c844dc64e0cab854997df9b3fd1b1c071ee25e6634e6b95bdb15f0acb38e7c5ac20d02f231a03fde3abb8220951328a7f550915ff9da4a1960c8005d6dfabc50f776bd17433f29e8d0566871380b439024a70dfade593173c6462ad49318-----
`
var signer = &mclsig.BlsSingleSigner{}
var keyGen = signing.NewKeyGenerator(&mcl.SuiteBLS12{})

func loadSkPk(t *testing.T) (crypto.PrivateKey, crypto.PublicKey, string){
	buff := []byte(blsPemContents)

	blkRecovered, _ := pem.Decode(buff)
	require.NotNil(t, blkRecovered)

	blockType := blkRecovered.Type
	header := "PRIVATE KEY for "
	require.True(t, strings.Index(blockType, header) == 0)
	blockTypeString := blockType[len(header):]

	skHex := blkRecovered.Bytes
	skBuff, err := hex.DecodeString(string(skHex))
	require.Nil(t, err)

	sk, err := keyGen.PrivateKeyFromByteArray(skBuff)
	require.Nil(t, err)

	pk := sk.GeneratePublic()
	pkBuff, err := pk.ToByteArray()
	require.Nil(t, err)
	require.Equal(t, blockTypeString, hex.EncodeToString(pkBuff))

	return sk, pk, blockTypeString
}

func sign(t *testing.T, message []byte, privKey crypto.PrivateKey) []byte{
	signature, err := signer.Sign(privKey, message)
	require.Nil(t, err)

	return signature
}

func verify(t *testing.T, message []byte, publicKey crypto.PublicKey, signature []byte){
	err := signer.Verify(publicKey, message, signature)
	require.Nil(t, err)
}

func TestBLSSigning(t *testing.T){
	privateKey, publicKey, publicKeyAsHex := loadSkPk(t)

	message := "message to be signed"
	signature := sign(t, []byte(message), privateKey)

	fmt.Printf("message: %s\npublicKey: %s\nsignature: %s\n\n",
		message,
		publicKeyAsHex,
		hex.EncodeToString(signature),
	)

	verify(t, []byte(message), publicKey, signature)
	fmt.Println("signature checked OK")
}


```

**Review by iulianpascalau — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]

**Pull request merged**